### PR TITLE
Refactor JVB Client Connection code

### PR DESCRIPTION
**Pull request by bbaldino**
This PR refactors the XMPP client connection code in the bridge to address a couple things:

1. XMPPCommon is leftover from when we supported both Component and Client connections and is no longer needed
2. Address a "deep" access of the ClientConnection singleton and move it to a higher layer as part of cleaning up the singleton code (see https://github.com/jitsi/jitsi-videobridge/issues/1444)

### CODE REVIEW

File: jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        if (running.compareAndSet(true, false)) {
+            mucClientManager.stop()
+        }
+    }
+
+    fun setPresenceExtension(extension: ExtensionElement) {
+        mucClientManager.setPresenceExtension(extension)
+    }
+
+    fun addMucClient(jsonObject: JSONObject): Boolean {
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt:82]: Can you bring back the docs from the original? They describe the json format

Comment (bgrozev) [reply jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt:82]: Other docs in this class are also useful

Comment (bbaldino) [reply jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt:82]: Oops, definitely.

File: jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    /**
+     * Process an incoming IQ
+     */
+    override fun handleIq(iq: IQ?): IQ? {
+        if (iq == null) {
+            return null
+        }
+        logger.cdebug { "RECV: ${iq.toXML()}" }
+
+        val response = when (iq.type) {
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt:122]: IQ types are `get`, `set`, `result` and `error`. The latter two are responses, and we wouldn't send anything back for them, so there's not need to log "SENT null".

The code could be simplified some, too: 
```
return when (iq.type) {
  IQ.Type.get, IQ.Type.set -> handleIqRequest(iq).also { log() }
  else -> null
}

File: jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    companion object {
+        private val FEATURES = arrayOf(
+            ColibriConferenceIQ.NAMESPACE,
+            HealthCheckIQ.NAMESPACE,
+            "urn:xmpp:jingle:apps:dtls:0",
+            "urn:xmpp:jingle:transports:ice-udp:1",
+            Version.NAMESPACE
+        )
+
+        private val delayThresholds = longArrayOf(5, 50, 100, 1000)
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt:186]: Note: these were static to accomodate both XMPP client and component modes. They don't need to be static anymore (but that doesn't need to change in this PR).

**Comment by bgrozev** [comment]
The name `ClientConnection` was used to distinguish from the XMPP component connection. I think this is the right time to rename it to `XmppConnection` or something that makes it more obvious what it is.

File: jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt (hunk 3 above)
Comment (bbaldino) [reply jvb/src/main/kotlin/org/jitsi/videobridge/xmpp/ClientConnection.kt:186]: Ok, yeah I think I'll leave those as-is for now.

**Pull request merged**